src/gurobi_optimods/opf/violations.py

```diff
@@ -139,17 +139,34 @@ def grbderive_xtra_sol_values_from_voltages(alldata, model):
         ]:
             constr = item[0]
             var = item[1]
-
-            row = model.getRow(constr)
-            sum = -constr.RHS
             leadcoeff = 0
-            for i in range(row.size()):
-                v = row.getVar(i)
-                coeff = row.getCoeff(i)
-                if v.Varname != var.Varname:
-                    sum += coeff * xbuffer[v]
-                else:
-                    leadcoeff = coeff
+            # linear constraint
```

Comment:
Could you please add some comments to describe what this is doing? It looks like this could all be replaced with a getValue() call, unless this is based on a different solution?

Reply:
Done